src/ripple/app/misc/Transaction.h

```diff
@@ -166,6 +278,18 @@ class Transaction
     TER             mResult = temUNCERTAIN;
     bool            mApplying = false;
 
+    /** different ways for transaction to be accepted */
+    bool            mApplied = false;
+    bool            mQueued = false;
+    bool            mBroadcast = false;
+    bool            mKept = false;
```

Comment:
Some of these values look mutually exclusive. Does it make sense to make an enum to represent the valid states instead of using the four bools here?

We should also look to see if some of these states are implied with the TER result. I.e, do we need a `mQueued` flag when we already have a `terQUEUED` `mResult`?

Reply:
I think that per-bit representation of these flags is better than as enum, because these flags come to us as bits and go to output as the same bits. In the case of mutually exclusive enum values we need to code input bits into enum values, then to decode it from enum values to bits. This will require more computational steps. The queued flag is preserved because it is mentioned in the original task for this ticket.